creusot/tests/should_succeed/iterators/03_std_iterators.coma

```diff
@@ -1422,62 +1414,58 @@ module M_03_std_iterators__counter [#"03_std_iterators.rs" 41 0 41 27]
   = (exists res_state : closure0'1 . postcondition_mut'0 self args res_state res /\ resolve'4 res_state)
   
   function unnest_trans'0 (self : closure0'1) (b : closure0'1) (c : closure0'1) : ()
   
-  axiom unnest_trans'0_spec : forall self : closure0'1, b : closure0'1, c : closure0'1 . ([%#sops37] unnest'0 self b)
-   -> ([%#sops38] unnest'0 b c)  -> ([%#sops39] unnest'0 self c)
+  axiom unnest_trans'0_spec : forall self : closure0'1, b : closure0'1, c : closure0'1 . ([%#sops36] unnest'0 self b)
+   -> ([%#sops37] unnest'0 b c)  -> ([%#sops38] unnest'0 self c)
   
   function unnest_refl'0 (self : closure0'1) : ()
   
-  axiom unnest_refl'0_spec : forall self : closure0'1 . [%#sops36] unnest'0 self self
+  axiom unnest_refl'0_spec : forall self : closure0'1 . [%#sops35] unnest'0 self self
   
   function postcondition_mut_unnest'0 (self : closure0'1) (args : (uint32, Snapshot.snap_ty (Seq.seq uint32))) (res_state : closure0'1) (res : uint32) : ()
     
   
-  axiom postcondition_mut_unnest'0_spec : forall self : closure0'1, args : (uint32, Snapshot.snap_ty (Seq.seq uint32)), res_state : closure0'1, res : uint32 . ([%#sops34] postcondition_mut'0 self args res_state res)
-   -> ([%#sops35] unnest'0 self res_state)
-  
-  let rec closure0'0 (_1:borrowed closure0'1) (x:uint32) (_prod:Snapshot.snap_ty (Seq.seq uint32)) (return'  (ret:uint32))= {[@expl:closure requires] [%#s03_std_iterators7] UIntSize.to_int ((_1.current).field_0'0).current
-    = Seq.length (Snapshot.inner _prod)
-    /\ ((_1.current).field_0'0).current < (v_MAX'0 : usize)}
-    (! bb0
-    [ bb0 = s0
-      [ s0 = UIntSize.add {((_1.current).field_0'0).current} {[%#s03_std_iterators6] (1 : usize)}
-          (fun (_ret':usize) ->
-             [ &_1 <- { _1 with current = { field_0'0 = { (_1.current).field_0'0 with current = _ret' } } } ] 
-            s1)
-      | s1 = -{resolve'0 _1}- s2
-      | s2 =  [ &res1 <- x ] s3
-      | s3 =  [ &res <- res1 ] s4
-      | s4 =  [ &_0 <- res ] s5
-      | s5 = return' {_0} ]
-       ]
-    )
-    [ & _0 : uint32 = any_l ()
-    | & _1 : borrowed closure0'1 = _1
-    | & x : uint32 = x
-    | & res : uint32 = any_l ()
-    | & res1 : uint32 = any_l () ]
-    
-    [ return' (result:uint32)-> {[@expl:closure ensures] [%#s03_std_iterators8] UIntSize.to_int ((_1.final).field_0'0).current
-      = UIntSize.to_int ((_1.current).field_0'0).current + 1
-      /\ UIntSize.to_int ((_1.final).field_0'0).current = Seq.length (Snapshot.inner _prod) + 1 /\ result = x}
-      {[@expl:closure unnest] unnest'0 _1.current _1.final}
-      (! return' {result}) ]
-    
+  axiom postcondition_mut_unnest'0_spec : forall self : closure0'1, args : (uint32, Snapshot.snap_ty (Seq.seq uint32)), res_state : closure0'1, res : uint32 . ([%#sops33] postcondition_mut'0 self args res_state res)
+   -> ([%#sops34] unnest'0 self res_state)
```

Comment:
I'm worried about these generated axioms:
- Are they valid with inference of specifications of closures?
- They do not seem valid evan before your PR, actually. Am I right?

It seems like we should always conjunct the postcondtion with `unnest`, just to make sure that this axiom is valid.

Reply:
Nevermind, I misread the code, sorry.